mpt_trie/src/trie_ops.rs

```diff
@@ -6,13 +6,41 @@ use std::{fmt::Display, mem::size_of};
 use enum_as_inner::EnumAsInner;
 use ethereum_types::{H256, U128, U256, U512};
 use log::trace;
+use thiserror::Error;
 
 use crate::{
     nibbles::{Nibble, Nibbles},
     partial_trie::{Node, PartialTrie, WrappedNode},
     utils::TrieNodeType,
 };
 
+/// Stores the result of trie operations. Returns a [TrieOpError] upon
+/// failure.
+pub type TrieOpResult<T> = Result<T, TrieOpError>;
+
+/// An error type for trie operation.
+#[derive(Debug, Error)]
+pub enum TrieOpError {
+    /// An error that occurs when a hash node is found during an insert
+    /// operation.
+    #[error("Found a `Hash` node during an insert in a `PartialTrie`! These should not be able to be traversed during an insert!")]
```

Comment:
@BGluth Are we happy with the error messages?

Reply:
Yeah I think the message is fine, but I think it's probably worth while also having the key piece where we encountered the hash node at (eg. `0x1234`).